syntax = "proto3";
import "google/protobuf/timestamp.proto";

message Tweet{
    int32 id = 1;
    string username = 2;
    string content = 3;
    google.protobuf.Timestamp posted_at = 4;
    google.protobuf.Timestamp last_edited_at = 5;
    repeated string tag = 6;
}

service TweetService{
    rpc GetAllTweets(Tweet) returns (stream Tweet) {}
    rpc GetTweets(Tweet) returns (stream Tweet) {}
    rpc CreateTweet(Tweet) returns (Tweet) {}
    rpc RemoveTweet(Tweet) returns (Tweet) {}
    rpc EditTweet(Tweet) returns (Tweet) {}
}